### PR TITLE
Fixes for commitment generation and forwarding

### PR DESCRIPTION
**Pull request by vgeddes**
Changes:
* Parachain
  * Fixed outdated format of ABI-encoded payload (necessary for correct commitment generation)
* Relayer
  * Updated contract bindings
  * Added documentation about how to update bindings
  * Forward commitment hash to Inbound channel
  * Use a static gas limit of `500000` instead of calling `eth_estimateGas`
* Solidity
  * Remove redundant gas checks

E2E tests pass successfully.

Closes: #268

### CODE REVIEW

File: relayer/chain/ethereum/writer.go [hunk 1]
```diff
@@ -105,9 +105,10 @@ func (wr *Writer) Write(ctx context.Context, msg *chain.SubstrateOutboundMessage
 	}
 
 	options := bind.TransactOpts{
-		From:    wr.conn.kp.CommonAddress(),
-		Signer:  wr.signerFn,
-		Context: ctx,
+		From:     wr.conn.kp.CommonAddress(),
+		Signer:   wr.signerFn,
+		Context:  ctx,
+		GasLimit: 500000,
```
Comment (musnit) [inline relayer/chain/ethereum/writer.go:111]: @vgeddes can we put this into an env/config file rather?

File: relayer/contracts/generate.go [hunk 2]
```diff
@@ -0,0 +1,4 @@
+//go:generate bash -c "jq .abi ../../ethereum/build/contracts/InboundChannel.json | abigen --abi - --type contract --pkg inbound --out inbound/contract.go"
+//go:generate bash -c "jq .abi ../../ethereum/build/contracts/OutboundChannel.json | abigen --abi - --type contract --pkg outbound --out outbound/contract.go"
+
+package contracts
```
Comment (philipstanislaus) [inline relayer/contracts/generate.go:4]: Nice!

File: ethereum/contracts/BasicInboundChannel.sol [hunk 3]
```diff
@@ -80,11 +62,8 @@ contract BasicInboundChannel is InboundChannel {
             nonce = nonce + 1;
 
             // Deliver the message to the target
-            // Delivery will have fixed maximum gas allowed for the target app
             (bool success, ) =
-                _messages[i].target.call{value: 0, gas: MAX_GAS_PER_MESSAGE}(
-                    _messages[i].payload
-                );
```
Comment (philipstanislaus) [inline ethereum/contracts/BasicInboundChannel.sol:87]: Maybe I'm missing something, but don't we still need this gas check? I think only the ones above with the `require` statements were unnecessary, right?

File: ethereum/contracts/IncentivizedInboundChannel.sol [hunk 4]
```diff
@@ -90,9 +64,7 @@ contract IncentivizedInboundChannel is InboundChannel {
             // Deliver the message to the target
             // Delivery will have fixed maximum gas allowed for the target app
             (bool success, ) =
-                _messages[i].target.call{value: 0, gas: MAX_GAS_PER_MESSAGE}(
-                    _messages[i].payload
-                );
```
Comment (philipstanislaus) [inline ethereum/contracts/IncentivizedInboundChannel.sol:95]: Likewise, I think only the checks above with the `require` statements were unnecessary, right?

File: ethereum/contracts/BasicInboundChannel.sol (hunk 3 above)
Comment (philipstanislaus) [reply ethereum/contracts/BasicInboundChannel.sol:87]: Nevermind, has been added here again: https://github.com/Snowfork/polkadot-ethereum/pull/279

File: ethereum/contracts/IncentivizedInboundChannel.sol (hunk 4 above)
Comment (philipstanislaus) [reply ethereum/contracts/IncentivizedInboundChannel.sol:95]: Nevermind, has been added here again: https://github.com/Snowfork/polkadot-ethereum/pull/279

**Pull request merged**